pandas/tests/io/json/test_pandas.py

```diff
@@ -82,76 +80,79 @@ def setup(self, datapath):
-        tm.assert_almost_equal(df.values, unser.values)
-        unser = read_json(df.to_json(orient="values"), orient="values")
-        tm.assert_numpy_array_equal(df.values, unser.values)
+    @pytest.mark.parametrize(
+        "orient,expected",
+        [
+            (
+                "split",
+                DataFrame([["a", "b"], ["c", "d"]], index=[1, 1], columns=["x", "y"]),
+            ),
+            ("records", DataFrame([["a", "b"], ["c", "d"]], columns=["x", "y"])),
+            ("values", DataFrame([["a", "b"], ["c", "d"]])),
+        ],
+    )
+    def test_frame_non_unique_index(self, orient, expected):
+        df = DataFrame([["a", "b"], ["c", "d"]], index=[1, 1], columns=["x", "y"])
 
-    def test_frame_non_unique_columns(self):
-        df = DataFrame([["a", "b"], ["c", "d"]], index=[1, 2], columns=["x", "x"])
+        result = read_json(df.to_json(orient=orient), orient=orient)
+        assert_frame_equal(result, expected)
 
-        msg = "DataFrame columns must be unique for orient='index'"
-        with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="index")
-        msg = "DataFrame columns must be unique for orient='columns'"
-        with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="columns")
-        msg = "DataFrame columns must be unique for orient='records'"
+    @pytest.mark.parametrize("orient", ["index", "columns"])
+    def test_frame_non_unique_index_raises(self, orient):
+        df = DataFrame([["a", "b"], ["c", "d"]], index=[1, 1], columns=["x", "y"])
+        msg = "DataFrame index must be unique for orient='{}'".format(orient)
         with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="records")
-
-        assert_frame_equal(
-            df, read_json(df.to_json(orient="split"), orient="split", dtype=False)
-        )
-        unser = read_json(df.to_json(orient="values"), orient="values")
-        tm.assert_numpy_array_equal(df.values, unser.values)
-
-        # GH4377; duplicate columns not processing correctly
-        df = DataFrame([["a", "b"], ["c", "d"]], index=[1, 2], columns=["x", "y"])
-        result = read_json(df.to_json(orient="split"), orient="split")
-        assert_frame_equal(result, df)
-
-        def _check(df):
-            result = read_json(
-                df.to_json(orient="split"), orient="split", convert_dates=["x"]
-            )
-            assert_frame_equal(result, df)
+            df.to_json(orient=orient)
 
-        for o in [
+    @pytest.mark.parametrize("orient", ["split", "values"])
+    @pytest.mark.parametrize(
+        "data",
+        [
             [["a", "b"], ["c", "d"]],
             [[1.5, 2.5], [3.5, 4.5]],
             [[1, 2.5], [3, 4.5]],
             [[Timestamp("20130101"), 3.5], [Timestamp("20130102"), 4.5]],
-        ]:
-            _check(DataFrame(o, index=[1, 2], columns=["x", "x"]))
+        ],
+    )
+    def test_frame_non_unique_columns(self, orient, data):
```

Comment:
pending https://github.com/pandas-dev/pandas/pull/27838/files?file-filters%5B%5D=#r324906825, this corresponds to L142-L154 and LGTM